lib/openhab/dsl/rules/automation_rule.rb

```diff
@@ -74,6 +74,9 @@ def on_removal(listener)
 
         # This method gets called in rspec's SuspendRules as well
         def execute!(mod, inputs)
+          # Store the context in a thread variable. It is accessed through DSL#method_missing
+          # which is triggered when the context variable is referenced inside the run block
+          @thread_locals[:openhab_context] = extract_context(inputs)
           ThreadLocal.thread_local(**@thread_locals) do
```

Comment:
instead of saving `extract_context` into @thread_locals, and having to remove it, simply pass it as an additional kwarg here:

```suggestion
          ThreadLocal.thread_local(openhab_context: extract_context(inputs), **@thread_locals) do
```

Reply:
It needed to be saved because it's needed again in #process_task, so if it's not saved into @thread_locals, it won't be there. I'll add a comment though.